spec/datetimeformat.html

```diff
@@ -286,43 +273,517 @@ <h1>Internal slots</h1>
+            </tr>
+            <tr>
+              <td>[[Pattern]]</td>
+              <td>a Pattern String</td>
+              <td>A String of the same format as the regular date pattern String.</td>
+            </tr>
+          </table>
+        </emu-table>
+      </emu-clause>
+
+      <emu-clause id="sec-datetimeformat-calendar-style-record">
+        <h1>DateTime Calendar Style Records</h1>
+
+        <p>
+          Each <dfn id="datetimeformat-calendar-style-record">DateTime Calendar Style Record</dfn> has the fields defined in <emu-xref href="#table-datetimeformat-calendar-style-record"></emu-xref>.
+        </p>
+        <emu-table id="table-datetimeformat-calendar-style-record">
+          <emu-caption>DateTime Calendar Style Record</emu-caption>
+          <table class="real-table">
+            <thead>
+              <tr>
+                <th>Field Name</th>
+                <th>Value Type</th>
+              </tr>
+            </thead>
+            <tr>
+              <td>[[DateFormat]]</td>
+              <td>a DateTime Style Record</td>
+            </tr>
+            <tr>
+              <td>[[TimeFormat]]</td>
+              <td>a DateTime Style Record</td>
+            </tr>
+            <tr>
+              <td>[[DateTimeFormat]]</td>
+              <td>a DateTime Connector Record</td>
+            </tr>
+            <tr>
+              <td>[[DateTimeRangeFormat]]</td>
+              <td>a DateTime Date Range Record</td>
+            </tr>
+          </table>
+        </emu-table>
+      </emu-clause>
+
+      <emu-clause id="sec-datetimeformat-style-record">
+        <h1>DateTime Style Records</h1>
+
+        <p>
+          Each <dfn id="datetimeformat-style-record">DateTime Style Record</dfn> has the fields defined in <emu-xref href="#table-datetimeformat-style-record"></emu-xref>.
+        </p>
+        <emu-table id="table-datetimeformat-style-record">
+          <emu-caption>DateTime Style Record</emu-caption>
+          <table class="real-table">
+            <thead>
+              <tr>
+                <th>Field Name</th>
+                <th>Value Type</th>
+                <th>Description</th>
+              </tr>
+            </thead>
+            <tr>
+              <td>[[full]]</td>
+              <td>a DateTime Format Record</td>
+              <td>Format record for the *"full"* style.</td>
+            </tr>
+            <tr>
+              <td>[[long]]</td>
+              <td>a DateTime Format Record</td>
+              <td>Format record for the *"long"* style.</td>
+            </tr>
+            <tr>
+              <td>[[medium]]</td>
+              <td>a DateTime Format Record</td>
+              <td>Format record for the *"medium"* style.</td>
+            </tr>
+            <tr>
+              <td>[[short]]</td>
+              <td>a DateTime Format Record</td>
+              <td>Format record for the *"short"* style.</td>
+            </tr>
```

Comment:
```suggestion
            <tr>
              <td>[[full]]</td>
              <td>a DateTime Format Record</td>
              <td>Format Record for the *"full"* style.</td>
            </tr>
            <tr>
              <td>[[long]]</td>
              <td>a DateTime Format Record</td>
              <td>Format Record for the *"long"* style.</td>
            </tr>
            <tr>
              <td>[[medium]]</td>
              <td>a DateTime Format Record</td>
              <td>Format Record for the *"medium"* style.</td>
            </tr>
            <tr>
              <td>[[short]]</td>
              <td>a DateTime Format Record</td>
              <td>Format Record for the *"short"* style.</td>
            </tr>
```

Reply:
Rebased again!  Most of the comments above are addressed by #887 -- I think the only one that isn't is the use of `id` in the DateTimeFormat Patterns Record `dfn`

@anba @gibson042